code/modules/food/drinkingglass/shaker.dm

```diff
@@ -19,19 +19,40 @@
 /obj/item/weapon/reagent_containers/food/drinks/glass2/fitnessflask/update_icon()
 	..()
 	icon_state = "[base_icon]_[lid_color]"
+	cut_overlays()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/pdrink.dmi', src, null)
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(0 to 9)
+				filling.icon_state = null
+				return
+			if(10 to 19) 	filling.icon_state = "[fillsource]10"
+			if(20 to 29) 	filling.icon_state = "[fillsource]20"
+			if(30 to 39) 	filling.icon_state = "[fillsource]30"
+			if(40 to 49) 	filling.icon_state = "[fillsource]30"
+			if(50 to 59) 	filling.icon_state = "[fillsource]50"
+			if(60 to 69) 	filling.icon_state = "[fillsource]60"
+			if(70 to 79) 	filling.icon_state = "[fillsource]70"
+			if(80 to INFINITY)	filling.icon_state = "[fillsource]80"
```

Comment:
Is there any good reason for this copypaste and not using normal filling_states, just with different numbers?